Lisandro

lisandromar opened the pull request:
Definido el struct del array de pokemones.
Ordenar los pokemones por tipo.
Dado el promedio de niveles de pokemones por tipo.

sopderbeck-wolox committed:
1) estructura del registro de los pokemones
2) cantidad de pokemones por tipo
3) promedio de niveles por tipo

sopderbeck-wolox committed:
solve indentation problem and add function to get information from file. add function to load information into file together the user load information into system

sebapderbeck commented on functions/Menu_controller.cpp:
```diff
@@ -30,15 +36,19 @@ void showNewPosition(int position);                     //prints arrows in curre
+void showMenuToEnterInformationFromUser(tPokemon rPokemon[], int &current_register_length);               //function get information from user, ask how many records do you want to load
+void showTemplateToEnterInformationFromUser(tPokemon rPokemon[], int current_register_length);  //function that assigns the information to the struct
 void showHeaderQuantityRemainsToEnter(int current_quantity, int quantity_limit_pokemons); //header that show how quantity remains to enter
+int getLimitPokemonInArray(); // pide al usuario un limite de pokemones para ingresar
```
esta funcion no se usa

lisandromar replied:
La saco, no es necesaria por el limite que dijo el profe.

sebapderbeck commented on functions/Menu_controller.cpp:
```diff
@@ -49,16 +59,22 @@ void showPokemonsGroupByLevel();
 void showPokemonWithLessLevel();
+void cantidadDePokemonesPorTipo(tPokemon rPokemon[], int current_register_length); //Da la cantidad de pokemones por tipo
+void mostrarCantidadDePokemonesPorTipo(int contador_tierra, int contador_fuego, int contador_agua, int contador_electrico); //muestra los contadores
+void promedioNivelPorTipo(tPokemon rPokemon[], int current_register_length); //calcula el promedio de niveles por tipo
```
el nombre de las funciones deberia estar en ingles

sebapderbeck commented on functions/Menu_controller.cpp:
```diff
@@ -49,16 +59,22 @@ void showPokemonsGroupByLevel();
     showLoadingBar();
-
+    int current_register_length = 0;
+    tPokemon rPokemon[802];
```
deberiamos poner el limite que dijo el profesor

sebapderbeck commented on functions/Menu_controller.cpp:
```diff
@@ -475,3 +511,95 @@ void showLoadingBar() {
     system("CLS");
 }
+
+int getLimitPokemonInArray(){
```
eliminaria esta funcion

sebapderbeck commented on functions/Menu_controller.cpp:
```diff
@@ -475,3 +511,95 @@ void showLoadingBar() {
+
+void cantidadDePokemonesPorTipo(tPokemon rPokemon[], int current_register_length){
+    int contador_tierra = 0;
+    int contador_fuego = 0;
```
unificaria la asignacion de todas estas variables para ocupar una sola linea

sebapderbeck commented on functions/Menu_controller.cpp:
```diff
@@ -475,3 +511,95 @@ void showLoadingBar() {
+        } else if (rPokemon[i].type == "a"){
+            contador_agua++;
+        } else {
+            contador_electrico++;
```
En varios lugares se calcula el total de pokemones de cada tipo, realizaria una funcion para unificar esto y llamarla cuando sea necesario

sebapderbeck commented on functions/Menu_controller.cpp:
```diff
@@ -475,3 +511,95 @@ void showLoadingBar() {
+    int contador_tierra = 0;
+    int contador_fuego = 0;
+    int contador_agua = 0;
+    int contador_electrico = 0;
```
unificar asignaciones

sebapderbeck commented on functions/Menu_controller.cpp:
```diff
@@ -475,3 +511,95 @@ void showLoadingBar() {
+    cout << " EL PROMEDIO DE NIVELES DE POKEMONES TIPO TIERRA ES: " << promedio_tierra;
+    gotoxy (32 , 24);
+    cout << " EL PROMEDIO DE NIVELES DE POKEMONES TIPO FUEGO ES: " << promedio_fuego;
+    gotoxy (32 , 26);
```
buscaria la forma de unificar la forma en la cual se imprimen los valores de las funciones resultantes. asi ahorramos lineas de codigo y hacemos que el programa se vea mas compacto.
por ejemplo en varios lugares se muestra a los pokemones divididos por tipo, unificaria estas vistas y agregaria un header para identificar cada funcion

sebapderbeck commented on functions/Menu_controller.cpp:
```diff
@@ -475,3 +511,95 @@ void showLoadingBar() {
+    gotoxy (32 , 26);
+    cout << " EL PROMEDIO DE NIVELES DE POKEMONES TIPO AGUA ES: " << promedio_agua;
+    gotoxy (32 , 28);
+    cout << " EL PROMEDIO DE NIVELES DE POKEMONES TIPO ELECTRICO ES: " << promedio_electrico;
```
por ejemplo
||PROMEDIO DE NIVELES DE POKEMONES ||
TIERRA:
FUEGO:
AGUA: 
ELECTRICO :
intentaria hacer algo asi pero mas lindo

sebapderbeck commented on functions/Menu_controller.cpp:
```diff
@@ -475,3 +511,95 @@ void showLoadingBar() {
+    int contador_agua = 0;
+    int contador_electrico = 0;
+    for (int i = 0; i < current_register_length; i++){
+        if (rPokemon[i].type == "t"){
```
el if deberia estar preguntando por la letra en mayuscula, ya que utilizamos una funcion para validar las letras y otra para tranformarla a uppercase

sebapderbeck commented on functions/Menu_controller.cpp:
```diff
@@ -388,30 +427,27 @@ void showMenuToEnterInformationFromUser() {
+void showTemplateToEnterInformationFromUser(tPokemon rPokemon[], int current_register_length) {
+    gotoxy (32 , 22);
+    cout << "   TIPO DE POKEMON: ";
+    cin >> rPokemon[current_register_length].type;
```
falta validacion del tipo

sebapderbeck commented on functions/Menu_controller.cpp:
```diff
@@ -388,30 +427,27 @@ void showMenuToEnterInformationFromUser() {
+
+    gotoxy (32 , 24);
+    cout << "   NIVEL DE POKEMON: ";
+    cin >> rPokemon[current_register_length].level;
```
falta validacion del nivel

sebapderbeck commented on functions/Menu_controller.cpp:
```diff
@@ -388,30 +427,27 @@ void showMenuToEnterInformationFromUser() {
+
+    gotoxy (32 , 26);
+    cout << "   NOMBRE DE POKEMON: ";
+    cin >> rPokemon[current_register_length].name;
```
falta validacion de nombre

sebapderbeck commented on functions/Menu_controller.cpp:
```diff
@@ -366,19 +382,42 @@ void goodbyeMessage() {
-    int lenght = 500;     //file size
-    char * buffer = new char [lenght]; //variable temporal
-    filePokedex.open("pokedex.dat", ios::in | ios::binary); //open file
+string ShowTemplateToEnterNameFromFile() {
```
falta agregar esta funcion y algunas otras al archivo de funciones (donde se explica que hace cada funcion)

sopderbeck-wolox committed:
add function that show header where you enter de information from file, refactor the code of function getInformationFromFile and ShowTemplateToEnterNameFromFile
the last one doesnt work because two reasons.
1) the funcion dont write the file in the correct form
2) the function dont read file

PROBLEMS IN FILES

sopderbeck-wolox committed:
Cambiadas las funciones a ingles, eliminada la que le pedia al usuario la cantidad de pokemones y puesto el limite de 50 como dijo el profe.

lisandromar committed:
Cambiadas las funciones a ingles, eliminada la que le pedia al usuario la cantidad de pokemones y puesto el limite de 50 como dijo el profe.

lisandromar committed:
Declaradas las variables en una sola linea.

lisandromar committed:
Llamadas las validaciones y creada la validacion del nombre

lisandromar committed:
Elimine la funcion getPokemonInArray()
el if ahora pregunta por las mayusculas

sopderbeck-wolox committed:
solve indentation problem and show message error implementation valid functions
